### PR TITLE
ITSASU-230: create new TextInputHelper

### PR DESCRIPTION
**Pull request by scullythornton1**

### CODE REVIEW

File: app/views/templates/PrincipalMainTemplate.scala.html [hunk 1]
```diff
@@ -21,7 +21,7 @@
         govukWrapper: GovUkWrapper
 )
 
-@(title: String, optForm: Option[Form[_]] = None, backLink: Option[String]= None)(contentBlock: Html)(implicit request: RequestHeader, messages: Messages, appConfig: AppConfig)
+@(title: String, optForm: Option[Form[_]] = None, backLink: Option[String]= None, showSignOutLink: Boolean = true)(contentBlock: Html)(implicit request: RequestHeader, messages: Messages, appConfig: AppConfig)
```
Comment (AlexRimmerHMRC) [inline app/views/templates/PrincipalMainTemplate.scala.html:24]: if you are adding this in, you should also add it below, I will add a comment with what I mean below

File: app/views/helpers/injected/TextInputHelper.scala.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+              ),
+                hint = hint.map{ text =>
+                     Hint(
+                        id = Some(s"${field.name}-hint"),
+                        content = Text(text)
+                    )
+                },
+                classes = classes.getOrElse("")
+        ).withFormField(field)
+)
```
Comment (AlexRimmerHMRC) [inline app/views/helpers/injected/TextInputHelper.scala.html:47]: The alignment of everything here looks out of place, can you reformat this again?

File: app/views/templates/PrincipalMainTemplate.scala.html (hunk 1 above)
Comment (scullythornton1) [reply app/views/templates/PrincipalMainTemplate.scala.html:24]: added

File: app/views/templates/GovUkWrapper.scala.html [hunk 3]
```diff
@@ -32,14 +32,14 @@
...
 }
 
 @beforeContentBlock = {
-@hmrcLanguageSelectHelper()
-@backLink.map {href =>
-    @govukBackLink(BackLink(href = href, content = Text(messages("base.back")), attributes = Map("id" -> "back-link")))
-}
+    @hmrcLanguageSelectHelper()
+    @backLink.map {href =>
+            @govukBackLink(BackLink(href = href, content = Text(messages("base.back")), attributes = Map("id" -> "back-link")))
```
Comment (AlexRimmerHMRC) [inline app/views/templates/GovUkWrapper.scala.html:41]:
```suggestion
        @govukBackLink(BackLink(href = href, content = Text(messages("base.back")), attributes = Map("id" -> "back-link")))
```

**Pull request merged**